src/rust/bitbox02-rust/src/hww/api/ethereum/signmsg.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2020 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::pb;
+use super::Error;
+
+use crate::workflow::verify_message;
+
+use pb::eth_response::Response;
+
+use alloc::vec::Vec;
+use core::convert::TryInto;
+
+use sha3::digest::Digest;
+
+/// Process a sign message request.
+///
+/// The result contains a 65 byte signature. The first 64 bytes are the secp256k1 signature in
+/// compact format (R and S values), and the last byte is the recoverable id (recid).
+pub async fn process(request: &pb::EthSignMessageRequest) -> Result<Response, Error> {
+    if request.msg.len() > 9999 {
+        return Err(Error::InvalidInput);
+    }
+    let pub_request = pb::EthPubRequest {
+        output_type: pb::eth_pub_request::OutputType::Address as _,
+        keypath: request.keypath.clone(),
+        coin: request.coin,
+        display: true,
+        contract_address: Vec::new(),
+    };
+
+    // Verify address. We don't need the actual result, but we have to propagate validation or user
+    // abort errors.
+    let _ = super::pubrequest::process(&pub_request).await?;
```

Comment:
I don't think you need `let _ =`. Cleaner imho. The comment obviously stays. It's good.

Also, I personally dislike hooking into other API from other entries points. Ideally in my world there would be a `crate::eth::verify_address(...)` which both `pubrequest::process` and this fn calls.

Reply:
did the first. postponing the second, it is not exactly clear to me how to do it.